### ISSUE TITLE
[FR]: Set an external tool(s) to open URLs from specific domains by default

### ISSUE THREAD

**Issue by RetroAbstract** [martinrotter/rssguard#844]
### Brief description of the feature request

This feature would allow the user to set links from one or multiple URL domains to be opened by a specific external tool by default when double clicking said links.

For example if the user wanted RSS Guard to automatically open Youtube links with the command prompt when double-clicking them in RSS Guard, and automatically open any other URLs in their article list with the default external browser they had already set, they would be able to do so by adding Youtube's domain in a "Domain(s)" prompt box following the already existing "Parameters" prompt when adding cmd.exe as an external tool.

In the external tools menu this would result in a "Domain(s)" column added, displaying what URL domains the user added to be associated with different external tools by default.

The user would be able to add multiple URL domains to be opened by an external tool by default by separating the URLs with commas.

**Comment by martinrotter** [comment]
Please, elaborate exactly how this should behave.

This feature should only trigger when any URL (or article URL) is launched in a specific way, not left-mbtn-clicked, but maybe middle button clicked, right?

**Comment by RetroAbstract** [comment]
> This feature should only trigger when any URL (or article URL) is launched in a specific way, not left-mbtn-clicked, but maybe middle button clicked, right?

No, as I explained further above, this would apply to regular article opening, either through left-mbtn double click or `Return` (or assigned hotkey to open an article with the external browser).

You would be able to assign one or more domains to an external tool so RSS Guard opens the assigned domains through the relevant external tool.


An example:

- I want to open YouTube URLs with [FreeTube](https://github.com/FreeTubeApp/FreeTube) when left-mbtn double clicking or pressing `Return` (or assigned hotkey to open an article with the external browser) on a selected article which URL starts with `youtube.com` instead of RSS Guard opening it through the external browser

To do this, I would enter the `youtube.com` domain in a "Domain" box in the FreeTube external tool line I create in the external tools settings page. RSS Guard now opens all articles which have `youtube.com` as base domain through FreeTube, rather than the external browser.

I'm struggling to elaborate this more than I already have.

A TL;DR of this feature: An exception list to the already existing "Open selected articles in external browser" feature (triggered by left-mbtn double clicking or pressing `Return` (or assigned hotkey to open an article with the external browser)) for specific article URL domains via the external tools. There would be no limit on how many URL domains you could assign to an external tool.

**Comment by martinrotter** [comment]
Yes, this is basically what I implemented. Check via devbuild please

**Comment by RetroAbstract** [comment]
Works perfect. Exactly what I hoped for. Thank you!
